### PR TITLE
Optimize images using gatsby fluid image

### LINKED ISSUE #8: Compress Images
Add a Github action workflow so Images are compressed so as to reduce the load time and network usage.

### PR DESCRIPTION
**Pull request by Korusuke**
Closes #8 

Test site for speed comparison: https://mlh-fellow-map-testing.surge.sh/
Have kept the resizing to `100x100` instead of `50x50` which is the actual size of icon as `50x50` has some pixelation on high res screens.

Also adds an exception for `gatsby-config.js` in eslint.

### CODE REVIEW

**Review by teikjun — APPROVED** [review]
Looks good to merge!

**Review by laveesh — APPROVED** [review]
:+1:

File: src/pages/index.tsx [hunk 1]
```diff
@@ -16,6 +16,12 @@ const DEFAULT_ZOOM = 3;
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (Ry-DS) [inline src/pages/index.tsx:16]: Reminder: We need to look into typing GraphQL responses. Maybe this markdown plugin has some predefined types for us.

**Review by Ry-DS — APPROVED** [review]
Definitely a lot faster! Is this fluid plugin built in? But since it works lets merge it :+1:

**Pull request merged**